### ISSUE TITLE
Smash Ultimate reproducible crash

### ISSUE THREAD

**Issue by ravenCondol** [Ryujinx/Ryujinx#1996]
### What's the issue you encountered?

Had 4 CPU's fighting each other in a Best of 5, random-stage gamerule set (to generate shader cache), but it always crashes when trying to return to main menu after set finish.

Only setting changed was enabling docked mode.

### How can the issue be reproduced?

Not sure if all this is necessary, but all I did was have 4 CPUs duke it out using following ruleset:

![Basic rule-set tab](https://puu.sh/Hdsqq/94d05e83a6.png)
![Advanced rule-set tab](https://puu.sh/HdsqV/f212398d99.png)

Then after one of them wins it crashes.

### Log file

Using LDN2.1 build:
[Ryujinx_1.0.0-ldn2.1_2021-02-03_17-45-21.log](https://github.com/Ryujinx/Ryujinx/files/5922890/Ryujinx_1.0.0-ldn2.1_2021-02-03_17-45-21.log)

Using latest (1.0.6471):
[Ryujinx_1.0.6471_2021-02-03_18-12-27.log](https://github.com/Ryujinx/Ryujinx/files/5922885/Ryujinx_1.0.6471_2021-02-03_18-12-27.log)

### Environment?

- Ryujinx version: Both **1.0.0-ldn2.1** and **1.0.6471**
- Game version: 10.1.0
- System Specs:
  - OS: Windows 10
  - CPU: Ryzen 5 **5600x** with base clock frequencies of 3700 Mhz, boost of up to 4700Mhz
  - GPU: NVIDIA RTX **3070** 8GB GDDR6 VRAM
  - RAM: 32GB of CL16 3600MHz G.SKILL **Ripjaws** V Series
- Applied Mods : None

### Additional context?

No, that is all.

**Comment by EmulationFanatic** [comment]
I was able to reproduce this crash using the above settings. 

```
00:15:05.986 |E| GUI.RenderLoop Cpu InvalidAccessHandler: Invalid memory access at virtual address 0x00000000E32E0000.
00:15:05.998 |E| GUI.RenderLoop Application : Unhandled exception caught: Ryujinx.Memory.InvalidMemoryRegionException: Attempted to access an invalid memory region.
   at Ryujinx.Memory.MemoryBlock.ThrowInvalidMemoryRegionException() in C:\projects\ryujinx\Ryujinx.Memory\MemoryBlock.cs:line 292
   at Ryujinx.Cpu.MemoryManager.ReadImpl(UInt64 va, Span`1 data) in C:\projects\ryujinx\Ryujinx.Cpu\MemoryManager.cs:line 461
   at Ryujinx.Cpu.MemoryManager.GetSpan(UInt64 va, Int32 size, Boolean tracked) in C:\projects\ryujinx\Ryujinx.Cpu\MemoryManager.cs:line 285
   at Ryujinx.Graphics.Gpu.Memory.Buffer.LoadRegion(UInt64 mAddress, UInt64 mSize) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Memory\Buffer.cs:line 316
   at Ryujinx.Memory.Tracking.MultiRegionHandle.QueryModified(UInt64 address, UInt64 size, Action`2 modifiedAction, Int32 sequenceNumber) in C:\projects\ryujinx\Ryujinx.Memory\Tracking\MultiRegionHandle.cs:line 111
   at Ryujinx.Graphics.Gpu.Memory.Buffer.SynchronizeMemory(UInt64 address, UInt64 size) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Memory\Buffer.cs:line 146
   at Ryujinx.Graphics.Gpu.Memory.BufferManager.CreateBufferAligned(UInt64 address, UInt64 size) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Memory\BufferManager.cs:line 514
   at Ryujinx.Graphics.Gpu.Memory.BufferManager.TranslateAndCreateBuffer(UInt64 gpuVa, UInt64 size) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Memory\BufferManager.cs:line 436
   at Ryujinx.Graphics.Gpu.Memory.BufferManager.SetComputeStorageBuffer(Int32 index, UInt64 gpuVa, UInt64 size, BufferUsageFlags flags) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Memory\BufferManager.cs:line 226
   at Ryujinx.Graphics.Gpu.Engine.Methods.Dispatch(GpuState state, Int32 argument) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Engine\Compute.cs:line 98
   at Ryujinx.Graphics.Gpu.Engine.GPFifo.GPFifoProcessor.Send(MethodParams meth) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Engine\GPFifo\GPFifoProcessor.cs:line 154
   at Ryujinx.Graphics.Gpu.Engine.GPFifo.GPFifoProcessor.Process(ReadOnlySpan`1 commandBuffer) in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Engine\GPFifo\GPFifoProcessor.cs:line 101
   at Ryujinx.Graphics.Gpu.Engine.GPFifo.GPFifoDevice.DispatchCalls() in C:\projects\ryujinx\Ryujinx.Graphics.Gpu\Engine\GPFifo\GPFifoDevice.cs:line 176
   at Ryujinx.Ui.GlRenderer.Render() in C:\projects\ryujinx\Ryujinx\Ui\GLRenderer.cs:line 404
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()
```

**Comment by gdkchan** [comment]
Should be fixed by #2083.
